### PR TITLE
[ Milestone 9 ]  Complete Project

### PR DESCRIPTION
**Pull request by CakeInTech**
- Implemented all the Group tasks.
- Implemented all the tasks for `Person1`, `Person2`, and `Person3`.
- [Here](https://drive.google.com/file/d/17ZOB384qqx1TsNvNRuaVKOJ6hpIOm9m6/view?usp=sharing) is Link to the project description video.

### CODE REVIEW

**Review by Ismailco — CHANGES_REQUESTED** [review]
#### Hi @MohamedCK, @Ceemos96, @jaamanu, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    end
+  end
+
+  def list_all_genre
+    genre = File.size('./json/genre.json').zero? ? [] : JSON.parse(File.read('./json/genre.json'))
+    puts '    '
+    genre.each do |genres|
+      puts "Name: #{genres['name']}"
+    end
+    puts '    '
```
[the remark this answers is not in the document]
Comment (MohamedCK) [reply app.rb:144]: Thanks for the reminder, I have included error handler for that case. :smile:

File: database/schema.sql [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    id SERIAL PRIMARY KEY,
+    genre_id INTEGER,
+    author_id INTEGER,
+    label_id INTEGER,
+    publish_date DATE,
+    archived BOOLEAN,
+    FOREIGN KEY(genre_id) REFERENCES genre(id)
+)
+
+
```
[the remark this answers is not in the document]
Comment (MohamedCK) [reply database/schema.sql:37]: Hi @Ismailco, It seems my coding parner had put the table in seperate file, so I will work on that.

File: database/schema.sql [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+CREATE TABLE labels (
+  id INTEGER GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+  title VARCHAR(100),
+  color VARCHAR(100)
+);
+
+CREATE TABLE musicalbum (
+  id SERIAL PRIMARY KEY,
+  on_spotify BOOLEAN,
+  publish_date Date
```
[the remark this answers is not in the document]
Comment (MohamedCK) [reply database/schema.sql:19]: Thanks for the reminder. I have included the `can_be_archived` column. :smile: :like:

**Review by prolajumokeoni — APPROVED** [review]
Hi @Team

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀  
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**